design/kiota-e2e.md

```diff
@@ -23,8 +23,155 @@ Before we jump into the end-to-end walk-through, it's important to set some cons
 
 ## Table of Contents
 
--   [NodeJS e2e using the Service library](#node-service-library)
--   [NodeJS e2e using the Core library](#node-core-library)
+- [Foudation](#foundation)
+- [NodeJS e2e using the Service library](#node-service-library)
+- [NodeJS e2e using the Core library](#node-core-library)
+
+## <a id="foundation"></a> Foundation
+
+This section lists all the foundation elements of the Graph JS SDK and how it affects the developer's experience.
+
+### Models
+
+The models should be easily consumable by developers in a natural format. For JavaScript and TypeScript developers, interfaces are the most common way to understand, represent and send data to endpoints.
+
+Thanks to the TypeScript nature of allowing merging of Classes and Interfaces with the same name, models are represented in a way that allows both rich capabilities of the classes (backing stores, dirty tracking, etc.) and simplicity of using interfaces. 
```

Comment:
Importing a class and interface with the same name gives me this issue.

Try the following:

```
//File a.ts
export interface name{
    id:string;
    test:string;
}

export class name{

    id:string;
    type:string;
    test:string;

    public getFieldDeserializers():Array<string>{
        return ["name","type"];
    }
}

```

```
//b.ts
import {name} from "./a";

const test: name ={ id:"name", type:"type"};

```

I get a prompt on the `test` variable that Error: Type '{ id: string; type: string; }' is missing the following properties from type 'name': test, getFieldDeserializersts(2739).

Interfaces can have duplicate names and that works fine in TypeScript because they do not transpile. However, we are not getting rid of the problem that we see today that requires all properties to be set when using a json initialization pattern because the class requires all the properties.

Reply:
We would need to think about the import pattern when using same named interfaces and classes.